ens/main.py

```diff
@@ -188,7 +190,7 @@ def resolve(self, name, get='addr'):
         else:
             return None
 
-    def resolver(self, normal_name):
+    def resolver(self, normal_name: str): # mt: Optional[ConciseContract]
```

Comment:
Yes, but... it will very likely the ens module, which is internally expecting the `ConciseContract` API. So maybe not here.

Also, I think the same concept will still exist, just not as a factory, with somewhat fewer features, and by a different name: `ContractCaller`. Updating ENS to use the new class can be a part of resolving #1025

Reply:
I liked `ContractReader` and exposing it via `Contract.reader` automatically as part of the core contract API.